packages/flutter/lib/src/cupertino/route.dart

```diff
@@ -302,8 +302,15 @@ class CupertinoPageRoute<T> extends PageRoute<T> {
   ) {
     if (route.fullscreenDialog) {
       return CupertinoFullscreenDialogTransition(
-        animation: animation,
+        primaryRouteAnimation: animation,
+        secondaryRouteAnimation: secondaryAnimation,
         child: child,
+        // Check if the route has an animation that's currently participating
+        // in a back swipe gesture.
+        //
+        // In the middle of a back gesture drag, let the transition be linear to
+        // match finger motions.
+        linearTransition: isPopGestureInProgress(route),
```

Comment:
I suspect this might not be needed. Can you find a native example in iOS where a fullscreen modal page is back swipable?

Reply:
Indeed the primaryanimation does not use this, however, the secondary animation does still needs it

Reply:
ah you're right, thanks